packages/react/src/popover/close/PopoverClose.tsx

```diff
@@ -31,7 +31,11 @@ export const PopoverClose = React.forwardRef(function PopoverClose(
     props: [
       {
         onClick(event) {
-          setOpen(false, createBaseUIEventDetails('close-press', event.nativeEvent));
+          setOpen(
+            false,
+            createBaseUIEventDetails('close-press', event.nativeEvent),
+            event.currentTarget,
```

Comment:
I'm curious about this signature. I think only one other component uses it, but it would probably be better if `eventDetails` could be extended with custom properties.

Reply:
I added the `trigger` field to BaseUIEventDetails and made Floating UI hooks aware of it as it'll be used more widely when we roll out detached triggers for other components.

Reply:
We'll have to align it with this PR afterwards: https://github.com/mui/base-ui/pull/2796